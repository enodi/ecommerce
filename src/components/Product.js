import React from "react";

const Product = ({ product, handleAddItemToCart, currency }) => (
  <div className="product__item">
    <img
      src={product.image_url}
      alt={product.title}
      className="product__image"
    />
    <p>{product.title}</p>
    <p>{`From: ${currency} ${product.price}.00`}</p>
    <button
      onClick={() => handleAddItemToCart(product)}
      className="product__cart"
    >
      Add to Cart
    </button>
  </div>
);

export default Product;
