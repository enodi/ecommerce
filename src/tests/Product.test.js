import React from "react";
import { shallow } from "enzyme";

import { cart } from "./fixtures";
import Product from "../components/Product";

let handleAddItemToCart, wrapper;
beforeEach(() => {
  handleAddItemToCart = jest.fn();
  wrapper = shallow(
    <Product
      handleAddItemToCart={handleAddItemToCart}
      currency="NGN"
      product={cart}
    />
  );
});

test("should render Product component correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

test("should find text rendered on buttom", () => {
  expect(wrapper.find("button").text()).toEqual("Add to Cart");
});

test("should add to cart", () => {
  wrapper.find("button").simulate("click");
  expect(handleAddItemToCart).toHaveBeenCalled();
});
