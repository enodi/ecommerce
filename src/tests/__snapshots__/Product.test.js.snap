// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Product component correctly 1`] = `
<div
  className="product__item"
>
  <img
    className="product__image"
  />
  <p />
  <p>
    From: NGN undefined.00
  </p>
  <button
    className="product__cart"
    onClick={[Function]}
  >
    Add to Cart
  </button>
</div>
`;
